contrib/natss/pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -77,10 +90,58 @@ func createReceiverFunction(s *SubscriptionsSupervisor, logger *zap.SugaredLogge
 }
 
 func (s *SubscriptionsSupervisor) Start(stopCh <-chan struct{}) error {
+	// Trigger Connect to establish connection with NATS
+	s.connect <- struct{}{}
 	s.receiver.Start(stopCh)
 	return nil
 }
 
+func (s *SubscriptionsSupervisor) connectWithRetry() {
```

Comment:
This should probably have a stop channel, even if only for the unit tests.

Reply:
done